firebase-storage/src/main/java/com/google/firebase/storage/GetDownloadUrlTask.java

```diff
@@ -71,7 +73,8 @@ private Uri extractDownloadUrl(JSONObject response) {
   @Override
   public void run() {
     final NetworkRequest request =
-        new GetMetadataNetworkRequest(storageRef.getStorageUri(), storageRef.getApp());
+        new GetMetadataNetworkRequest(
+            storageRef.getStorageUri(), storageRef.getApp(), storageRef.getEmulatorSettings());
```

Comment:
Can `getStorageUri` return an URL that includes the Emulator? You will not have to expose `StorageReference#getEmulatorSettings` and you will likely not have to change much in the NetworkRequest classes.

Reply:
That URI is always a `gs://` URI now so I'm not sure where I could stuff in the emulator details.

For example it would be something like `gs://foo.appspot.com/child` and the code makes the assumption that `getAuthority()` returns `foo.appspot.com` and `getPath()` returns `child`.

Do you think it's safe to turn this into a non-gs URI?  Seems like a risky change.

Reply:
I tried to play with it. It seems much cleaner but it is also a fair bit of work. This is a draft state: https://gist.github.com/schmidt-sebastian/950943121b49b6542f9b4fceda54c540

I gave up after 45 minutes or so, but if you feel like this is worth pursuing you or I could pick it back up. Otherwise, we can probably leave it as is.

Reply:
🙏 thank you! You always end up sending me a patch file when I get your review :-)